pkg/util/etcdutil/migrate.go

```diff
@@ -209,7 +209,13 @@ func waitBootEtcdRemoved(etcdServiceIP string) error {
 }
 
 func cleanupBootstrapEtcdService(kubecli kubernetes.Interface) {
-	if err := kubecli.CoreV1().Services(api.NamespaceSystem).Delete(bootstrapEtcdServiceName, &v1.DeleteOptions{}); err != nil {
-		glog.Errorf("failed to remove bootstrap-etcd-service: %v", err)
+	if err := wait.Poll(pollInterval, pollTimeout, func() (bool, error) {
+		if err := kubecli.CoreV1().Services(api.NamespaceSystem).Delete(bootstrapEtcdServiceName, &v1.DeleteOptions{}); err != nil {
+			glog.Errorf("failed to remove bootstrap-etcd-service: %v", err)
```

Comment:
only nit might be to check if this was a "does not exist" error. But not sure that's actually a situation we could realistically get into.